uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -78,65 +78,84 @@
 
 use anyhow::{bail, Result};
 
-use super::record::Field;
+use super::record::{Field, FieldDescr};
 use super::types::Type;
-use super::{APIConverter, ComponentInterface};
+use super::{APIConverter, ComponentInterface, CINode};
 
 /// Represents an enum with named variants, each of which may have named
 /// and typed fields.
 ///
 /// Enums are passed across the FFI by serializing to a bytebuffer, with a
 /// i32 indicating the variant followed by the serialization of each field.
+#[derive(Debug)]
+pub struct Enum<'a> {
+    pub(super) parent: &'a ComponentInterface,
+    pub(super) descr: &'a EnumDescr,
+}
```

Comment:
So the `Enum` struct is now a kind of "fat pointer" thing that contains both a pointer to the data and parent ref to the containing object.